web/src/components/network/DnsDataList.tsx

```diff
@@ -55,36 +66,36 @@ export default function DnsDataList({ servers: originalServers, updateDnsServers
     updateDnsServers(servers);
   };
 
-  const updateServer = (id, field, value) => {
+  const updateServer = (id: number, field: string, value: string) => {
     const server = servers.find((dns) => dns.id === id);
     server[field] = value;
     updateDnsServers(servers);
   };
 
-  const deleteServer = (id) => {
+  const deleteServer = (id: number) => {
     const serverIdx = servers.findIndex((dns) => dns.id === id);
     servers.splice(serverIdx, 1);
     updateDnsServers(servers);
   };
 
-  const renderDns = ({ id, address }) => {
+  const renderDns = ({ id, address }: DNS) => {
```

Comment:
hmm..when I check DNS type it can have optional id...what happens when id is not in object in this case?

Reply:
When missing, we're adding it at line 60 when creating an internal DNS collection based on the original ones. This id/index is needed to be used as key and avoid the component messing all up when the user adds and removes servers. Related to https://react.dev/learn/rendering-lists#why-does-react-need-keys.

Something that can be improved, for sure, but completely out of scope of this PR. This component somehow inherits a behavior we implemented for cockpit-widget long ago.